bindings/go/oci/integration/integration_test.go

```diff
@@ -129,6 +129,113 @@ func Test_Integration_OCIRepository_BackwardsCompatibility(t *testing.T) {
 	})
 }
 
+func Test_Integration_HealthCheck_Authentication(t *testing.T) {
+	if testing.Short() {
+		t.Skipf("skipping integration test as reaching ghcr.io is taking too long!")
+	}
+
+	t.Parallel()
+	user, password := getUserAndPasswordWithGitHubCLIAndJQ(t)
+
+	r := require.New(t)
+	r.NotEmpty(user)
+	r.NotEmpty(password)
+
+	reg := "ghcr.io"
+
+	t.Run("health check with valid authentication succeeds", func(t *testing.T) {
+		r := require.New(t)
+
+		resolver, err := urlresolver.New(urlresolver.WithBaseURL(reg))
+		r.NoError(err)
+		resolver.SetClient(createAuthClient(reg, user, password))
+
+		repo, err := oci.NewRepository(oci.WithResolver(resolver))
+		r.NoError(err)
+
+		// Health check should succeed with valid credentials
+		err = repo.CheckHealth(t.Context())
+		r.NoError(err)
+	})
+
+	t.Run("health check with invalid authentication fails", func(t *testing.T) {
+		r := require.New(t)
+
+		resolver, err := urlresolver.New(urlresolver.WithBaseURL(reg))
+		r.NoError(err)
+		resolver.SetClient(createAuthClient(reg, "invalid-user", "invalid-password"))
+
+		repo, err := oci.NewRepository(oci.WithResolver(resolver))
+		r.NoError(err)
+
+		// Health check should fail for ghcr.io with invalid credentials
+		// GHCR returns 403 during token exchange when credentials are invalid
+		err = repo.CheckHealth(t.Context())
+		r.Error(err)
+		r.True(strings.Contains(err.Error(), "403") || strings.Contains(err.Error(), "denied"),
```

Comment:
So this is a bit shit. Apparently, I can't just ping ghcr... It immediately redirects to a token exchange that results in a a 403.

Reply:
Technically, it can return a 401 too.
```
➜ curl -I -X GET https://ghcr.io/v2/
HTTP/2 401
content-type: application/json
docker-distribution-api-version: registry/2.0
strict-transport-security: max-age=63072000; includeSubDomains; preload
www-authenticate: Bearer realm="https://ghcr.io/token",service="ghcr.io",scope="repository:user/image:pull"
date: Wed, 02 Jul 2025 14:17:29 GMT
content-length: 73
x-github-request-id: E086:25DF22:CB3F9:CD484:68653F79

```

But apparently, oras isn't getting that.

Reply:
the get request you are receiving is correct because your curl doesnt have a token.
the ping needs to have the token set

Reply:
Basically what im saying is that the check health failure you receive here is completely fine because the invalid credentials make it unhealthy. for me this is okay. WDYTß

Reply:
Meh, maybe it's fine. I can't do anything about it. It's inside ORAS.

Reply:
hmm thats interesting. you might want to check if the authentication client is setup correctly.